b310-digital/mindwendel
# Allow customization of label names

Closes #72

How to test the feature:
- Checkout branch
- `mix ecto.migrate`
- `mix phx.server`
- Go to an old brainstorming; it should still be accessible and labelled ideas should still be labelled

If you want to test the correct migration of existing labels, you will need to do the following:
- `mix ecto.rollback --step 3` rollback migrations added by this pull request
- Open a console `iex -S mix`
- `Code.require_file("test/support/factory.ex")`
- `alias Mindwendel.Factory`
- `b = Factory.insert!(:brainstorming, %{labels: []})`
- `i = Factory.insert!(:idea, %{brainstorming: b, label_old: :label_1})`
- Then execute the migration with `mix ecto.migrate`
- Then `mix phx.server`
- Open brainstorming with `b.id`
- You should now see the brainstorming with one labelled idea

ToDos:
- [x] Fix styling for checked idea label
- [x] Fix button "Sort by Label" to consider order position
- [x] Add test for sort by label
- [x] Integrate label migration script in migration
- [x] Test label migration manually
- [x] Embed idea label order position in migration script
- [x] Consider i18n for labels
- [ ] Add new `schema.sql`
- [x] Added new screenshots
- [x] Move `field :label_old` to the migration file
- [x] "save labels" button is missing a translation
- [x] feedback is missing, after you click "save labels". Do we really need a separate button here? cant we do it during the save of the brainstorming?
- [x] Adjust the feature section in the `README.md`
- [x] Fixed styling bug regarding the wrong icons
- [x] Fixed styling bug regarding wrong alignment of icons

<img width="1904" alt="image" src="https://user-images.githubusercontent.com/592424/111114632-c8ad7700-8563-11eb-93eb-47eb9f71bfce.png">
